### PR TITLE
BadgerDB support

### PR DESCRIPTION
**Pull request by ondrej-smola**
Support for BadgerDB as and alternative to BoltDB based on #22. Open for review and discussion.

https://github.com/dgraph-io/badger

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ The goal of `valkeyrie` is to abstract common store operations (Get/Put/List/etc
 
 This repository is a fork of the [docker/libkv](https://github.com/docker/libkv) project which includes many fixes/additional features and is maintained by an original project maintainer. This project is notably used by [containous/traefik](https://github.com/containous/traefik), [docker/swarm](https://github.com/docker/swarm) and [docker/libnetwork](https://github.com/docker/libnetwork).
 
-As of now, `valkeyrie` offers support for `Consul`, `Etcd`, `Zookeeper`, `Redis` (**Distributed** store) and `BoltDB` (**Local** store).
+As of now, `valkeyrie` offers support for `Consul`, `Etcd`, `Zookeeper`, `Redis` (**Distributed** store) , `BoltDB` (**Local** store) and `BadgerDB` (**Local** store).
```
Comment (victorcoder) [inline README.md]: We can follow the same convention as Distributed ones and just have a single `**Local** store` at the end

Comment (victorcoder) [reply README.md]: I think BadgerDB is just Badger without the DB

File: store/store.go [hunk 2]
```diff
@@ -24,6 +24,8 @@ const (
 	REDIS Backend = "redis"
 	// DYNAMODB backend
 	DYNAMODB Backend = "dynamodb"
+	// BADGERDB backend
+	BADGERDB Backend = "badgerdb"
```
Comment (victorcoder) [inline store/store.go]:
```suggestion
	BADGER Backend = "badger"
```

File: store/badger/badger_test.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
+package badgerdb
```
Comment (victorcoder) [inline store/badger/badger_test.go]:
```suggestion
package badger
```

**Comment by ondrej-smola** [comment]
Thanks, I think BadgerDB is the same as BoltDB. First line on project page says "BadgerDB is an embeddable, persistent ..." - i am ok with both ways, choosed *DB to be similar to BoltDB. Will fix ci/readme.

**Comment by victorcoder** [comment]
@ondrej-smola Sorry, I missed the mentions with BadgerDB you are right, I'm also ok either way, so let it as is. Thanks

**Comment by ondrej-smola** [comment]
@victorcoder please retry CI check, it seems like it failed due to some internal error

**Comment by abronan** [comment]
Coverage decreased thus the red but we can still merge it after review (-1.8% seems minor and an artifact of an additional backend being added).

**Comment by ondrej-smola** [comment]
@abronan I am already integrating this project into our solution - will add additional tests when we do some real-world testing and find any problems.

**Comment by abronan** [comment]
@ondrej-smola Very last step, but would it be possible to squash and rebase the MR on master? We'll be ready to merge then. Thanks!

**Comment by victorcoder** [comment]
@ondrej-smola ping :)

**Comment by abronan** [comment]
We can also carry this from there I think :)

**Comment by ldez** [comment]
Hello,

we changed the core system of Valkeyrie: now the stores have dedicated repositories.

So this PR must be converted to a dedicated repository.

To create the repo, you have to use our [repository template](https://github.com/kvtools/template)
And if you want to can open a [new issue](
https://github.com/kvtools/valkeyrie/issues/new?assignees=&labels=&template=new_store.yml) to ask for the transfer of your repository to the [kvtools organization](https://github.com/kvtools).

**Pull request closed without merging**